chain/indexer/src/streamer/mod.rs

```diff
@@ -135,13 +135,18 @@ pub(crate) async fn start(
     let mut last_synced_block_height: Option<near_primitives::types::BlockHeight> = None;
 
     'main: loop {
-        time::delay_for(INTERVAL).await;
-        let status = fetch_status(&client).await;
-        if let Ok(status) = status {
-            if status.sync_info.syncing {
-                continue;
+        match indexer_config.await_for_node_synced {
+            AwaitForNodeSyncedEnum::WaitForFullSync => {
+                time::delay_for(INTERVAL).await;
```

Comment:
@frol I'm not sure about putting delay only in await for node to be synced case or in any case.